### ISSUE TITLE
Custom UID/GID mapping for SFTP Server

### ISSUE THREAD

**Issue by bennoleslie** [ronf/asyncssh#74]
It would be useful to provide some way of extending the UID->username, GID->groupname mapping to occurs within format_longname.

Basic wish would be for an extension point of some sort over this code:

```
        user = _get_user_name(name.attrs.uid)
        group = _get_group_name(name.attrs.gid)
```

Now, I can overwrite the `format_longname` method, but then I copy everything except those two lines, which isn't nice.

I can probably also monkeypatch `_get_user_name`/`_get_group_name`, but that is also undesirable. If there was a way to pass a user/group dictionary into the SFTPServer constructor. Or simply make `_get_user_name` and `_get_group_name` methods then they can be overwritten instead of the full `format_longname` method.

I'm happy to put together a patch for whichever approach is deemed best; I don't have any strong opinions on the exact approach.

**Comment by ronf** [comment]
Good idea.

I've gone ahead and moved the code in _get_user_name() and _get_group_name() into overridable methods on the SFTPServer class. By default, they'd do exactly what the current top-level functions do, but you can replace them if you want to provide a custom mapping and the default format_longname() method will use your replacement.

This is now checked into the "develop" branch.

**Comment by ronf** [comment]
This feature is now available in the 1.9.0 release.
